src/client/components/dashboard/NarrativeList/ControlMenu/RevertNarrative.tsx

```diff
@@ -0,0 +1,249 @@
+import React, { Component } from 'react';
+import { getCurrentUserPermission } from '../../../../utils/narrativeData';
+import ControlMenuItemProps from './ControlMenuItemProps';
+import DashboardButton from '../../../generic/DashboardButton';
+import { LoadingSpinner } from '../../../generic/LoadingSpinner';
+import Runtime from '../../../../utils/runtime';
+import {
+  KBaseServiceClient,
+  KBaseDynamicServiceClient,
+} from '@kbase/narrative-utils';
+import { RouteComponentProps, withRouter } from 'react-router';
+import ControlMenu from './ControlMenu';
+
+type ComponentStatus =
+  | 'none'
```

Comment:
I think the `none` status is unused. You might prefer to omit it and make the `loading` state the default.